src/dotnet/APIView/APIViewWeb/Languages/LanguageProcessor.cs

```diff
@@ -42,34 +44,67 @@ public override async Task<CodeFile> GetCodeFileAsync(string originalName, Strea
             {
                 await stream.CopyToAsync(file);
             }
+            return await RunParserProcess(originalName, tempDirectory, jsonFilePath, Name);
+        }
 
-            try
+        public string RunProcess(string workingDirectory, string processName, string arguments)
+        {
+            var processStartInfo = new ProcessStartInfo(ProcessName, arguments);
+            string processErrors = String.Empty;
+
+            processStartInfo.WorkingDirectory = workingDirectory;
+            processStartInfo.RedirectStandardError = true;
+            processStartInfo.RedirectStandardOutput = true;
+            processStartInfo.UseShellExecute = false;
+            processStartInfo.CreateNoWindow = true;
+
+            var output = new StringBuilder();
+            var error = new StringBuilder();
+
+            using (var process = new Process())
             {
-                var arguments = GetProcessorArguments(originalName, tempDirectory, jsonFilePath);
-                var processStartInfo = new ProcessStartInfo(ProcessName, arguments);
-                string processErrors = String.Empty;
+                process.StartInfo = processStartInfo;
+                process.OutputDataReceived += (sender, args) =>
+                {
+                    if (args.Data != null)
+                        output.AppendLine(args.Data);
+                };
 
-                processStartInfo.WorkingDirectory = tempDirectory;
-                processStartInfo.RedirectStandardError = true;
-                processStartInfo.RedirectStandardOutput = true;
+                process.ErrorDataReceived += (sender, args) =>
+                {
+                    if (args.Data != null)
+                        error.AppendLine(args.Data);
+                };
+                process.Start();
+                process.BeginOutputReadLine();
+                process.BeginErrorReadLine();
+                process.WaitForExit();
```

Comment:
Do we need to pass a timeout here? What happens if the process hangs and never exits?